don't override fov when it's set to 0

Berthalamew opened the pull request:
- players shouldn't be able to set it to 0 within the menus but add check for people who set it manually.
- remove text in config for the hiresfix
- set default fov to 78 instead of 70 in the config (we already have the default option in imgui set to 78)

nukeulater commented:
> set default fov to 78 instead of 70 in the config (we already have the default option in imgui set to 78)

Then either set the default configuration to xbox or to h2v vanilla values, that including the crosshair offset.
Also add the missing upper FOV limit that was removed for some reason and split this in multiple commits. (one for removing the res fix leftovers, etc.)

Berthalamew commented:
Since you brought it up the crosshair offset shouldn't even be 0.138 in the first place. Vista by default uses 0.165, the same as original xbox.

Berthalamew committed:
remove config text for old high res fix

Berthalamew committed:
add upper fov limit

- don't override the fov if someone has it set to 0
- set the default fov to 78 to parody original xbox

Berthalamew committed:
change default crosshair offset to 0.165

- 0.165 is the default value on both h2x and h2v

Berthalamew committed:
Revert "change default crosshair offset to 0.165"

This reverts commit 0cce29cfee421b4c88238c3c1a17c1dfe447e03c.

Berthalamew committed:
Merge remote-tracking branch 'upstream/development-new' into hud_definition

Berthalamew committed:
add function to set character type for local users in network session

- use for loop to loop through all local players instead of just player 0
- add enum e_controller_index

Berthalamew commented:
Reverted back to 0.138 since even though the value is the same, the actual in-game offset is different. 
0.138 mimics the value on h2 xbox.

nukeulater commented on xlive/Blam/Engine/input/controllers.h:
```diff
@@ -0,0 +1,11 @@
+
+enum e_controller_index : int
+{
+	controller_index_0 = 0,
```
I don't see the point of this enum

nukeulater commented on xlive/Blam/Engine/Networking/Session/NetworkSession.h:
```diff
@@ -231,10 +233,10 @@ CHECK_STRUCT_SIZE(s_player_identifier, 8);
 	s_player_identifier network_user_identifier;
 	BYTE gap_B[3];
-	DWORD controller_index;
+	e_controller_index controller_index;
```
just do `int`

nukeulater commented on xlive/H2MOD/Modules/PlayerRepresentation/PlayerRepresentation.cpp:
```diff
@@ -112,7 +113,12 @@ namespace PlayerRepresentation
 			if (H2Config_spooky_boy && !Memory::IsDedicatedServer())
-				*Memory::GetAddress<e_character_type*>(0x51A67C) = character_type_skeleton;
+			{
+				for (byte user_index = 0; user_index <= k_number_of_users; ++user_index)
```
out of bounds...

nukeulater replied:
also doesn't follow the existing coding style

`for (int <name>; <name> < <condition>; <name>++)`

Berthalamew replied:
Forgot to change the <= to a < when swapping from the controller index enum to user indexes in these loops. 
Fixed

Berthalamew committed:
fix function definitions to use user_index instead of e_controller_index
